src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Collections/IntervalTree`1.cs

```diff
@@ -35,24 +36,69 @@ internal partial class IntervalTree<T> : IEnumerable<T>
 
     protected Node? root;
 
-    public static IntervalTree<T> Create<TIntrospector>(in TIntrospector introspector, IEnumerable<T>? values1 = null, IEnumerable<T>? values2 = null)
+    public static IntervalTree<T> Create<TIntrospector>(in TIntrospector introspector, IEnumerable<T>? values = null)
         where TIntrospector : struct, IIntervalIntrospector<T>
     {
         var result = new IntervalTree<T>();
 
-        AddAll(in introspector, values1);
-        AddAll(in introspector, values2);
+        if (values != null)
+        {
+            foreach (var value in values)
```

Comment:
>foreach (var value in values) [](http://example.com/codeflow?start=12&length=29)

Is sorting and calling into CreateFromSorted not possible?

Reply:
I want to do that in follow-up (also where I get rid of trees)